src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagEntry.cs

```diff
@@ -44,5 +46,21 @@ public override string ToString()
         {
             return $"Path: {Path}, VR:{VR}, Level:{Level} Status:{Status}";
         }
+
+        public override int GetHashCode()
+        {
+            return HashCode.Combine(Path, VR, Level.GetHashCode(), Status.GetHashCode());
+        }
+
+        public override bool Equals(object obj)
```

Comment:
>Equals [](start = 29, length = 6)

nit: I'd implement `IEquatable<CustomTagEntry>` if you're already going to override `Equals`